### PR TITLE
Padding mask invariance of Standardize

### PR DESCRIPTION
**Pull request by han-ol**
Addresses #726.

### CODE REVIEW

File: bayesflow/networks/helpers/standardization/standardize.py [hunk 1]
```diff
@@ -145,14 +151,21 @@ def _update_moments(self, x: Tensor, index: int):
...
-        batch_count = keras.ops.cast(keras.ops.prod(keras.ops.shape(x)[:-1]), self.count[index].dtype)
-
-        # Compute batch mean and M2 per feature
-        batch_mean = keras.ops.mean(x, axis=reduce_axes)
-        batch_m2 = keras.ops.sum((x - expand_left_as(batch_mean, x)) ** 2, axis=reduce_axes)
+        if mask is None:
+            mask = keras.ops.ones(keras.ops.shape(x)[:-1], dtype=x.dtype)
+        else:
+            mask = keras.ops.cast(keras.ops.broadcast_to(mask, keras.ops.shape(x)[:-1]), x.dtype)
+        mask = mask[..., None]
```
Comment (stefanradev93) [inline bayesflow/networks/helpers/standardization/standardize.py]: Multiplying with ones seems wasteful for large tensors unless we make a leap of faith that the compiler will optimize it away.

**Comment by stefanradev93** [comment]
That would work, but the mask needs to be passed by approximators to standardizer too.

File: bayesflow/networks/helpers/standardization/standardize.py [hunk 2]
```diff
@@ -25,6 +25,10 @@ def __init__(self, **kwargs):
         self.moving_m2 = None
         self.count = None
 
+    def compute_mask(self, inputs, mask=None):
+        # The mask is consumed here (moment computation). Nothing is propagated downstream.
+        return None
+
     def moving_std(self, index: int) -> Tensor:
```
Comment (arrjon) [inline bayesflow/networks/helpers/standardization/standardize.py:34]: why do we need this?

Comment (han-ol) [reply bayesflow/networks/helpers/standardization/standardize.py:34]: The mask argument is magic in keras and this is the way to tell it to not warn about not being able to inject mask arguments in later keras.layers.

Comment (han-ol) [reply bayesflow/networks/helpers/standardization/standardize.py:34]: I added a commit so the inline comment explains it better. The keras docs are lacking on this. It is however a real behavior. I found out about it by seeing the warning and then looking at keras source code.

The alternatives to defining such methods in each class are:

* excepting the confusing warning
* use a different keyword arg name other than `mask`. But not sure what would be a good candidate.
* outsource it to a `MaskConsuming` mixin class. Then, this is not repeated across multiple classes. This might be my favorite.

What do you think of all this?
